### DISCUSSION TITLE
Ping Support Websockets?

### DISCUSSION THREAD

**Discussion by Toorero** [encode/starlette#2623] [Q&A] [answered]
I'm looking for a way to do the following things:
- specify ping interval to keep the websocket alive
- specify ping timeout after which a websocket will be closed

I was scrubbing through the [websocket code](https://github.com/encode/starlette/blob/master/starlette/websockets.py) but couldn't find any explicit mentioning or support of the [Ping and Pong frames specified in RFC 6455](https://datatracker.ietf.org/doc/html/rfc6455#section-5.5.2). But then I tested the websocket implementation with `websocat --ping-interval 1 --ping-timeout 2 -E --text ws://localhost:8080/ -` and the connection didn't close, so there must be some support. What am I missing?

**Comment by Kludex** [comment, marked as answer]
The ping pong is handled by the server.

**Comment by Toorero** [comment]
So I can't send a ping as a client with starlett to keep the connection open nor can I specify the timeout?

**Reply by Kludex** [reply]
With starlette you can't.

**Reply by Toorero** [reply]
I see. That's too bad. Thanks for the reply though <3

**Reply by Kludex** [reply]
Not really... That's the job of the server, not the web framework.
